site/en/docs/privacy-sandbox/shared-storage/index.md

```diff
@@ -0,0 +1,219 @@
+   selected.
+*  [**Noisy aggregation of cross-site data**](#aggregated-data): you will be
+   able to run a worklet script to send your data through the
+   [Private Aggregation API](https://github.com/alexmturner/private-aggregation-api),
+   a Privacy Sandbox proposal, which returns a privacy-preserving report. 
+
+
+### Example: URL selection for frequency capping {: #url-selection }
+
+To select and create an opaque URL, register a worklet module to read shared
+storage data. The worklet class receives a list of up to eight URLs and then
+returns the index of the chosen URL. 
+
+When the client calls `sharedStorage.runURLSelectionOperation()`, the worklet
+executes and returns an opaque URL to be rendered into a fenced frame.
+
+Let's say you want to render an ad based on the advertiser's frequency cap
+(the maximum number of impressions for a user on a single ad). The frequency
+cap value is stored in shared storage. The shared storage worklet reads the
+values in shared storage, and decrements the value with each additional view.
+If there are available impressions left (the user has not hit their frequency
+cap), the ad is returned (index `1`). If not, the default URL is returned
+(index `0`).
+
+In this example, the adtech creates two files:
+
+*  `get-url.js` is the shared storage worklet, which evaluates the frequency
+   cap to determine which URL is returned for the ad creative.
+*  `main.js` selects and renders the ad creative and is injected on a
+   publisher's website.
+
+**get-url.js**
+
+```javascript
+class SelectURLOperation {
+  async run(data, urls) {
+
+    // Read the frequency cap value in shared storage
+    const frequencycap = await this.sharedStorage.get('frequencycap');
+
+    // If no impressions remain, return the default URL
+    if (frequencyCap === 0) {
+      return 0;
+    }
+
+    // If there are available impressions, subtract 1 and return the ad URL
+    await this.sharedStorage.set('frequencycap', frequencycap - 1);
+    return 1;
+  }
+}
+
+// Register the operation
+registerURLSelectionOperation(
+   'select-url', SelectURLOperation
+);
+```
+
+**main.js**
+
+```javascript
+// Set up the default and ad URLs
+const defaultUrl = new URL('https://default.example');
+const adUrl = new URL('[https://ad.example](https://ad.example)');
+
+// Register the worklet module
+await window.sharedStorage.worklet.addModule(‘get-url.js');
+
+// Set the frequency cap for this campaign, if it wasn't already set
+window.sharedStorage.set('frequencycap', 5, { ignoreIfPresent: true });
+
+// Select the URL from available options
+const opaqueUrl = await window.sharedStorage.runURLSelectionOperation(
+  'select-url', [defaultUrl, adUrl]
+);
+
+// Render the returned URL into an iframe or a fenced frame
+document.getElementById('example-iframe').src = opaqueUrl;
+```
+
+
+### Example: aggregation of cross-site data (ad campaign reach) {: #aggregated-data }
```

Comment:
Just leaving a reaction:  Oh, I didn't realize that the explainer example was updated: https://github.com/pythagoraskitty/shared-storage/commit/8ec100d7ee0dae1c52fb0a53f040b30af89053f9.  Yea, this example is a much better example.  Way better than what I attempted to come up with.